cmd/kube-apiserver/app/server.go

```diff
@@ -241,6 +243,25 @@ func Run(s *options.ServerRunOptions) error {
 		}
 	}
 
+	client, err := internalclientset.NewForConfig(genericConfig.LoopbackClientConfig)
+	if err != nil {
+		kubeAPIVersions := os.Getenv("KUBE_API_VERSIONS")
+		if len(kubeAPIVersions) == 0 {
+			return fmt.Errorf("failed to create clientset: %v", err)
+		}
+
+		// KUBE_API_VERSIONS is used in test-update-storage-objects.sh, disabling a number of API
+		// groups. This leads to a nil client above and undefined behaviour further down.
+		// TODO: get rid of KUBE_API_VERSIONS or define sane behaviour if set
+		glog.Errorf("Failed to create clientset with KUBE_API_VERSIONS=%q. KUBE_API_VERSIONS is only for testing. Things will break.", kubeAPIVersions)
+	}
+
+	// NOTE(ericchang): Internal informers should switch to using
+	// 'pkg/client/informers/informers_generated', the second informer
+	// created here.
+	sharedInformers := informers.NewSharedInformerFactory(nil, client, 10*time.Minute)
+	internalSharedInformers := internalversion.NewSharedInformerFactory(client, 10*time.Minute)
```

Comment:
This is how often a full resync occurs. A full resync is defined as the shared informer's DeltaFIFO cache re-enqueing every item it has so all listeners can have an opportunity to reprocess them. The correct Resync() function is here (or, the equivalent in the kubernetes repo): https://github.com/kubernetes/client-go/blob/204f12b1f3125cc660f989655af4b5bd83cbd9dd/tools/cache/delta_fifo.go#L507